cloudcix_primitives/hyperv.py

```diff
@@ -1,31 +1,858 @@
+        3046: f'Failed to unmount primary storage at {domain_path}{domain}\\mount on Host {host}',
+        3047: f'Failed to connect the Host {host} for the payload delete_mount_dir',
+        3048: f'Failed to delete mount dir {domain_path}{domain}\\mount on Host {host}',
+        3049: f'Failed to connect the Host {host} for the payload create_domain',
+        3050: f'Failed to create domain {domain} on Host {host}',
+        3051: f'Failed to connect the Host {host} for the payload set_cpu',
+        3052: f'Failed to set cpu {cpu} to domain {domain} on Host {host}',
+        3053: f'Failed to connect the Host {host} for the payload set_ram',
+        3054: f'Failed to set ram {ram}MB to domain {domain} on Host {host}',
+        3055: f'Failed to connect the Host {host} for the remove_default_nic',
+        3056: f'Failed to remove default nic from domain {domain} on Host {host}',
+        3057: f'Failed to connect the Host {host} for the add_gateway_vlan',
+        3058: f'Failed to add gateway vlan {gateway_vlan} to domain {domain} on Host {host}',
+        3059: f'Failed to connect the Host {host} for the add_secondary_vlans',
+        3060: f'Failed to add secondary vlans to domain {domain} on Host {host}',
+        3061: f'Failed to connect the Host {host} for the add_secondary_storages',
+        3062: f'Failed to add secondary storages to domain {domain} on Host {host}',
+        3063: f'Failed to connect the Host {host} for the start_domain',
+        3064: f'Failed to start domain {domain} on Host {host}',
+    }
 
+    messages_list = []
 
-def scrubprep():
-    return(False, 'Not Implemted')
+    # validate primary_storage
+    def validate_primary_storage(ps, msg_index):
+        if ps is None:
+            messages_list.append(f'{messages[msg_index]}: {messages[msg_index]}')
+            return False
+        if type(primary_storage) is not str:
+            messages_list.append(f'{messages[msg_index + 1]}: {messages[msg_index + 1]}')
+            return False
 
+        ps_items = ps.split('.')
+        if len(ps_items) != 2:
+            messages_list.append(f'{messages[msg_index + 2]}: {messages[msg_index + 2]}')
+            return False
+        elif ps_items[1] not in ('vhd', 'vhdx'):
+            messages_list.append(f'{messages[msg_index + 3]}: {messages[msg_index + 3]}')
+            return False
+        return True
 
-def updatequiesced():
-    return(False, 'Not Implemted')
+    validated = validate_primary_storage(primary_storage, 3011)
 
+    # validate secondary storages
+    def validate_secondary_storages(sstgs, msg_index):
+        if type(sstgs) is not list:
+            messages_list.append(f'{messages[msg_index]}: {messages[msg_index]}')
+            return False
 
-def updaterunning():
-    return(False, 'Not Implemted')
+        errors = []
+        valid_sstgs = True
+        for storage in secondary_storages:
+            if type(storage) is not str:
```

Comment:
Is this necessary, payload uses f-strings which ensures all sent is in string format